vmm/src/lib.rs

```diff
@@ -329,35 +329,36 @@ impl Vmm {
         }
 
         // Now we can boot the VM.
-        if let Some(ref mut vm) = self.vm {
-            vm.boot()
+        if self.vm.is_some()
```

Comment:
There is no need to change to this new pattern, is there?

`self.vm` is still going to be of either `Some(X)` or `None`.

You can shorten the following line quite a bit. That makes things easier to reason about.